plugins/flashrom/fu-flashrom-lspcon-i2c-spi-device.c

```diff
@@ -141,17 +222,9 @@ fu_flashrom_lspcon_i2c_spi_device_set_version (FuDevice *device, GError **error)
 		return FALSE;
 	}
 
-	/* grab the active partition */
-	self->active_partition = contents[entries[0].start];
-
-	if (self->active_partition != 1 && self->active_partition != 2) {
-		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_READ,
-				     "failed to read active partition");
-		return FALSE;
-	}
-
-	/* find the current version for the active partition */
-	addr = entries[self->active_partition + 2].start;
+	/* extract the active partition's version number */
+	addr = layout->entries[self->active_partition + 2].start;
+	g_return_val_if_fail(addr < (flash_size - 2), FALSE);
```

Comment:
i'd be much happier with a g_set_error here if it's a plausible runtime problem.

Reply:
The partition number is already validated and the address is static, so this is just defensive against getting the addresses wrong (which would be a bug).

Reply:
then space before bracket please.